modules/gmake/gmake_cpp.lua

```diff
@@ -521,7 +521,7 @@ end
 
 
 	function make.includes(cfg, toolset)
-		local includes = toolset.getincludedirs(cfg, cfg.includedirs, cfg.sysincludedirs)
+		local includes = toolset.getincludedirs(cfg, cfg.includedirs, cfg.sysincludedirs, cfg.frameworkdirs)
```

Comment:
Should probably be done for all exporters.

Reply:
This "miss" was intentional. I don't think resflags needs framework dirs.

Reply:
For resources, it might be OK, but I would pass `nil` explicitly.

For gmake2, only for "per-file" configuration has been done.
Missing for Codelite generator...

Reply:
I am not against an explicit nil, however there is currently no `toolset.getincludedirs(cfg, inc, nil)` when sysdir is not mandatory.
About gmake2, Ok, I didn't understand your first message quite well. This will be fixed.
About CodeLite, I said in the PR message I am not using CodeLite, have no knowledge about it and I can't test it.

Reply:
`sysdir` is missing only for resources flags (I wonder it is not a bug BTW).

You should add `frameworkdirs` for all non-resources flags call (for all generators).

It is the toolset which "manages" it. (as msvc which does nothing).
if generator doesn't use directly compiler flags, it iterates directly on `cfg.includedirs` or similar, and then indeed, it would require knowledge of the generator.

For test, I started https://github.com/Jarod42/premake-sample-projects to test with projects with multiple generators.
Nothing platform specific yet, as `frameworks` 

If you can provide sample test project (usable in github action) (a variant of project-01 which test includedirs), I can add it.

Reply:
@Jarod42 It is okay if contributors only want to target a single toolset. Not everyone has access to every toolset; allowing people to contribute where they can is more important. And please do not confuse things by asking people to provide you with tests for a separate repository; unit tests are all that we require at this time. Thanks!

Reply:
@starkos: What bothers me is that, then projects are generator-specific without any errors/warnings for users when used with incompatible generators.

Sorry, I didn't want to confuse people with extra tests in another repository.

Notice that you have tests just for gcc, but not for gmake/gmake2.

Reply:
> What bothers me is that, then projects are generator-specific without any errors/warnings for users when used with incompatible generators.

The alternative is that people don't contribute. We've been down that path before. I do agree that we need to do a better job of surfacing compatibility information, and the documentation seems an appropriate place to do that.

@Jarod42 If you have general comments on broad topics ("features should be implemented for all generators") please use the provided Discussions rather than cluttering up the PR review notes. I have no longer have any idea what change was requested by this thread, and if it has been done or not.